lib/main.dart

```diff
@@ -0,0 +1,364 @@
+  final GlobalKey<ScaffoldMessengerState> _scaffoldMessengerKey = GlobalKey();
+  final ValueNotifier<List<ChatMessage>> _messageNotifier = ValueNotifier([]);
+  final ValueNotifier<List<ParticipantId>> _participantListUpdatedNotifier = ValueNotifier([]);
+
+  @override
+  void initState() {
+    super.initState();
+    widget.callClient
+      ..setUsername(widget.prefs.getString('username') ?? 'Guest')
+      ..updateSubscriptionProfiles(
+        forProfiles: {
+          SubscriptionProfile.base: const MediaSubscriptionSettingsUpdate.set(
+            camera: CameraSubscriptionSettingsUpdate.set(
+              subscriptionState: SubscriptionStateUpdate.unsubscribed,
+            ),
+          ),
+          SubscriptionProfile.activeRemote: const MediaSubscriptionSettingsUpdate.set(
+            camera: CameraSubscriptionSettingsUpdate.set(
+              subscriptionState: SubscriptionStateUpdate.subscribed,
+            ),
+          ),
+        },
+      )
+      ..updateInputs(
+        inputs: const InputSettingsUpdate.set(
+          camera: CameraInputSettingsUpdate.set(isEnabled: BoolUpdate.set(true)),
+          microphone: MicrophoneInputSettingsUpdate.set(isEnabled: BoolUpdate.set(true)),
+        ),
+      );
+    _eventSubscription = widget.callClient.events.listen(_handleEvent);
+  }
+
+  @override
+  void setState(VoidCallback fn) {
+    super.setState(fn);
+    final participantIds = widget.callClient.participants.remote.keys.whereNot(_participantFocusPriority.contains);
+    if (!const IterableEquality().equals(participantIds, _participantListUpdatedNotifier.value)) {
+      _participantListUpdatedNotifier.value = [...participantIds];
+    }
+  }
+
+  void _handleEvent(Event event) {
+    if (!mounted) return;
+    event.whenOrNull<void>(
+      callStateUpdated: (callStateData) {
+        setState(() {
+          final callState = callStateData.state;
+          if (callState == CallState.joining) {
+            unawaited(Wakelock.toggle(enable: true));
+          } else if (callState == CallState.left) {
+            _participantFocusPriority.clear();
+            _messageNotifier.value = [];
+            unawaited(Wakelock.toggle(enable: false));
+          }
+          _hasActiveCall = callState == CallState.joining || callState == CallState.joined;
+        });
+      },
+      activeSpeakerChanged: (participant) {
+        if (!_hasActiveCall ||
+            participant == null ||
+            participant.info.isLocal ||
+            participant.id == _participantFocusPriority.firstOrNull) return;
+        _activeSpeakerToSet = participant.id;
+        if (_participantFocusPriority.length < 2) {
+          _updateActiveSpeaker();
+        } else {
+          _setActiveSpeakerTimer ??= Timer(const Duration(seconds: 2), _updateActiveSpeaker);
+        }
+      },
+      participantJoined: (participant) {
+        if (!_hasActiveCall || participant.info.isLocal || _participantFocusPriority.length >= 3) {
```

Comment:
Just a question on the event here - usually "participant-joined" events are only for remote participants, which would make an `isLocal` check redundant. This event name sounds similar, so just wanted to clarify - will this get fired for local participants too?

Reply:
I'll fix this. This code was originally written during a period when daily-core had a bug that made the `isLocal`-field unreliable.